submissions/xT0PERx/OOP exercise/index.js

```diff
@@ -0,0 +1,82 @@
+class Mammal {
+  constructor(name, gender, friend, saying, species, legs) {
+    this.name = name;
+    this.gender = gender;
+    this.friend = friend;
+    this.saying = saying;
+    this.species = species;
+    this.legs = legs;
+  }
+  toString() {
+    return [
+      this.name,
+      this.gender,
+      this.friend,
+      this.saying,
+      this.species,
+      this.legs,
+    ].join(";");
+  }
+}
+
+class Human extends Mammal {
+  constructor(
+    name,
+    gender,
+    friend,
+    saying,
+    species = "human",
+    legs = 2,
+    hands = 2
+  ) {
+    super(name, gender, friend, saying, species, legs);
+    this.hands = hands;
+  }
+  toString() {
+    return [super.toString(), this.hands].join(";");
+  }
+}
+
+class Cat extends Mammal {
+  constructor(name, gender, saying, friend, species = "cat", legs = 4) {
+    super(name, gender, friend, saying, species, legs);
+  }
+  toString() {
+    return super.toString();
+  }
+}
+class Dog extends Mammal {
+  constructor(name, gender, saying, species = "dog", friend, legs = 4) {
+    super(name, gender, friend, saying, species);
+    this.legs = legs;
+  }
+  toString() {
+    return super.toString();
+  }
```

Comment:
Why do we need to overload this method?

Reply:
You are right - there is no need.

Reply:
@xT0PERx
Why do we need to call `toString()` in constructors of `Dog` and `Cat`?
Why other classes' constructors do not call `toString()`?
Is this function defined at all?

Please do not consider the questions above rhetoric. I want to understand the decision making process behind.

Reply:
I call this method in the _"Dog"_ and _"Cat"_ classes in order to display their properties.
Constructors of other classes call the "toString()" method, but with additional properties that the superclass does not have.
>toString() {
    return [super.toString(), this.hands].join(';');
  }

The function toString() is defined in the superclass Mammals.

Reply:
It turns out like. I believe that overriding the "toString()" method from the Object methods in the superclass Mammals - helps me to use
>print (inhabitant));

Because otherwise I will be displayed on the screen __[Object, Object]__.
It just happened that I missed the point that I no longer need to call the method __"toString()"__ in __Dog__ and __Cat__, because he already does everything I need in the Mammals superclass

Reply:
@xT0PERx please revisit OOP principles. You need to understand inheritance better.